tests/testing_utils.py

```diff
@@ -0,0 +1,74 @@
+"""Utilities for running tests"""
+
+import string
+import random
+import numpy as np
+
+from git_theta import metadata
+
+
+def random_oid():
+    return "".join([random.choice(string.hexdigits.lower()) for _ in range(64)])
+
+
+def random_commit_hash():
+    return "".join([random.choice(string.hexdigits.lower()) for _ in range(40)])
+
+
+def random_lfs_metadata():
+    version = random.choice(["lfs_version1", "my_version", "version1"])
+    oid = "".join([random.choice(string.hexdigits.lower()) for _ in range(64)])
+    size = str(random.randint(0, 10000))
+    return metadata.LfsMetadata(version=version, oid=oid, size=size)
+
+
+def random_tensor_metadata():
+    ndims = random.choice(range(1, 6))
+    shape = tuple([random.choice(range(1, 50)) for _ in range(ndims)])
+    tensor = np.random.rand(*shape)
+    return metadata.TensorMetadata.from_tensor(tensor)
+
+
+def random_theta_metadata():
+    update_type = random.choice(["dense", "sparse"])
+    last_commit = "".join([random.choice(string.hexdigits.lower()) for _ in range(40)])
+    return metadata.ThetaMetadata(update_type=update_type, last_commit=last_commit)
+
+
+def random_param_metadata():
+    tensor_metadata = random_tensor_metadata()
+    lfs_metadata = random_lfs_metadata()
+    theta_metadata = random_theta_metadata()
+    return metadata.ParamMetadata(
+        tensor_metadata=tensor_metadata,
+        lfs_metadata=lfs_metadata,
+        theta_metadata=theta_metadata,
+    )
+
+
+def random_metadata():
```

Comment:
Lets do a quick refactor on this and the `make_nested_dict` utility.

1. Move `make_nested_dict` from `utils_test` to here.
2. Probably rename it `random_nested_dict`
3. Update the signature to take args like `possible_keys` and `possible_values` with defaults from the original `make_nested_dict` (convert to tuples or make the default values `None` and do a check in the function.
4. Update `random_metadata` to call `random_nested_dict` with the correct possible keys and values and then convert the returned value into a Metadata object.